src/hotspot/share/gc/shenandoah/shenandoahControlThread.cpp

```diff
@@ -481,8 +485,9 @@ void ShenandoahControlThread::handle_alloc_failure(ShenandoahAllocRequest& req,
 
 void ShenandoahControlThread::handle_alloc_failure_evac(size_t words) {
   ShenandoahHeap* heap = ShenandoahHeap::heap();
+  bool is_humongous = words > ShenandoahHeapRegion::humongous_threshold_words();
 
-  if (try_set_alloc_failure_gc()) {
+  if (try_set_alloc_failure_gc(is_humongous)) {
```

Comment:
I think we can replace this with assert(words <= ShenandoahHeapRegion::humongous_ghreshold_words()

We do not evacuate humongous objects.